subsystems/networking/bluetooth.go

```diff
@@ -0,0 +1,698 @@
+func initializeReadOnlyBluetoothCharacteristic(bsl *bluetoothServiceLinux, cName string, encoding uint16) bluetooth.CharacteristicConfig {
+	cUUID := bluetooth.NewUUID(uuid.New()).Replace16BitComponent(encoding)
+	bsl.logger.Debugf("%s can be read from the following bluetooth characteristic: %s", cName, cUUID.String())
+	c := &bluetooth.Characteristic{}
+
+	// refreshAvailableWiFiNetworks represents the callback function used
+	// to dynamically update an otherwise read-only bluetooth characteristic.
+	bsl.refreshAvailableWiFiNetworks = func(bs []byte) error {
+		_, err := c.Write(bs)
+		return err
+	}
+	return bluetooth.CharacteristicConfig{
+		Handle: c,
+		UUID:   cUUID,
+		Flags:  bluetooth.CharacteristicReadPermission,
+	}
+}
+
+// initializeBluetoothService performs low-level system configuration to enable bluetooth advertisement.
+func initializeBluetoothService(deviceName string, characteristics []bluetooth.CharacteristicConfig,
+) (bluetooth.UUID, *bluetooth.Advertisement, error) {
+	adapter := bluetooth.DefaultAdapter
+	if err := adapter.Enable(); err != nil {
+		return [4]uint32{}, nil, fmt.Errorf("failed to enable bluetooth adapter: %w", err)
+	}
+	serviceUUID := bluetooth.NewUUID(uuid.New()).Replace16BitComponent(0x1111)
+	if err := adapter.AddService(
+		&bluetooth.Service{
+			UUID:            serviceUUID,
+			Characteristics: characteristics,
+		},
+	); err != nil {
+		return [4]uint32{}, nil, fmt.Errorf("unable to add bluetooth service to default adapter: %w", err)
+	}
+	if err := adapter.Enable(); err != nil {
+		return [4]uint32{}, nil, fmt.Errorf("failed to enable bluetooth adapter: %w", err)
+	}
+	defaultAdvertisement := adapter.DefaultAdvertisement()
+	if defaultAdvertisement == nil {
+		return [4]uint32{}, nil, errors.New("default advertisement is nil")
+	}
+	if err := defaultAdvertisement.Configure(
+		bluetooth.AdvertisementOptions{
+			LocalName:    deviceName,
+			ServiceUUIDs: []bluetooth.UUID{serviceUUID},
+		},
+	); err != nil {
+		return [4]uint32{}, nil, fmt.Errorf("failed to configure default advertisement: %w", err)
+	}
+	return serviceUUID, defaultAdvertisement, nil
+}
+
+const (
+	BluezDBusService  = "org.bluez"
+	BluezAgentPath    = "/custom/agent"
+	BluezAgentManager = "org.bluez.AgentManager1"
+	BluezAgent        = "org.bluez.Agent1"
+)
+
+// validateBlueZVersion retrieves the installed BlueZ version via D-Bus.
+func validateBlueZVersion() error {
+	var versionOutput bytes.Buffer
+	var err error
+
+	// Try to get version from bluetoothctl first, fallback to bluetoothd
+	versionCmds := []string{"bluetoothctl --version", "bluetoothd --version"}
+	for _, cmd := range versionCmds {
+		versionOutput.Reset() // Clear buffer
+		cmdParts := strings.Fields(cmd)
+		execCmd := exec.Command(cmdParts[0], cmdParts[1:]...) //nolint:gosec
+		execCmd.Stdout = &versionOutput
+		err = execCmd.Run()
+		if err == nil {
+			break // Found a valid command
+		}
+	}
```

Comment:
Somewhat minor for now, but I do think it'd be better to test this via D-BUS (and during initBluetooth()) in order to really validate that you're talking to bluez in the expected way. Running commands may not always work as expected, and even if they're installed, it doesn't mean the service is actually running.

Also, there's an easier way to just run a one-shot command without setting up your own buffers or anything.
```
cmd := exec.Command("bluetoothctl", "--version")
output, err := cmd.CombinedOutput()
```